### ISSUE TITLE
Should renderOnDomContentLoaded check document.readyState?

### ISSUE THREAD

**Issue by sjrd** [raquo/Laminar#175]
Hi,

When compiling a Laminar application to Wasm, `renderOnDomContentLoaded` silently "does nothing". This is because Wasm apps are loaded through a top-level `await`, which typically means that when the `main` method executes, the DOM has *already* loaded.

If a `"DOMContentLoaded"` listener is registered after the DOM has already loaded, the event never gets fired.

[MDN recommends](https://developer.mozilla.org/en-US/docs/Web/API/Document/DOMContentLoaded_event#checking_whether_loading_is_already_complete) to check as follows:

> ```javascript
> function doSomething() {
>   console.info("DOM loaded");
> }
> 
> if (document.readyState === "loading") {
>   // Loading hasn't finished yet
>   document.addEventListener("DOMContentLoaded", doSomething);
> } else {
>   // `DOMContentLoaded` has already fired
>   doSomething();
> }
> ```

Indeed, I was able to work around the issue by doing the following in my `main` method in Scala.js:
```scala
    if dom.document.readyState == "loading" then
      renderOnDomContentLoaded(dom.document.body, p("hello 1"))
    else
      render(dom.document.body, p("hello 2"))
```
When compiling to Wasm, `"hello 2"` shows up.

Should we integrate that test directly into Laminar's `renderOnDomContentLoaded` so that others don't get confused the way I was? Something like:
```diff
   @inline def renderOnDomContentLoaded(
     container: => dom.Element,
     rootNode: => nodes.ReactiveElement.Base
   ): Unit = {
-    documentEvents(_.onDomContentLoaded).foreach { _ =>
-      new RootNode(container, rootNode)
-    }(unsafeWindowOwner)
+    if (dom.document.readyState == "loading") {
+      documentEvents(_.onDomContentLoaded).foreach { _ =>
+        new RootNode(container, rootNode)
+      }(unsafeWindowOwner)
+    } else {
+      new RootNode(container, rootNode)
+    }
   }
```

**Comment by raquo** [comment]
Oh, good catch, thanks! Lucky that we have this `renderOnDomContentLoaded` helper method to hide the non-standard condition in.

**Comment by Quafadas** [comment]
As a note, I believe that this also happens if one tries to use Laminar inside an mdoc:js code fence. 

sjrd's solution above seems to fix it.

**Comment by raquo** [comment]
@jchapuis Yes, the fix is already implemented, I'll include it in the upcoming v18. That version will have a bunch of other stuff that isn't fully done yet, ETA Q4 2025.
